pydantic/json_schema.py

```diff
@@ -0,0 +1,768 @@
+from __future__ import annotations
+
+import re
+from dataclasses import is_dataclass
+from enum import Enum
+from ipaddress import IPv4Address, IPv4Interface, IPv4Network, IPv6Address, IPv6Interface, IPv6Network
+from pathlib import Path
+from typing import Any, Callable, NewType, Pattern, cast
+from uuid import UUID
+
+from pydantic_core import CoreSchema, CoreSchemaType, core_schema
+from pydantic_core.core_schema import TypedDictField
+from typing_extensions import TypeGuard
+
+from pydantic._internal._core_metadata import CoreMetadataHandler
+from pydantic._internal._typing_extra import all_literal_values, is_namedtuple
+from pydantic._internal._utils import lenient_issubclass
+from pydantic.json import pydantic_encoder
+from pydantic.json_schema_misc import JsonSchemaValue
+from pydantic.networks import IPvAnyAddress, IPvAnyInterface, IPvAnyNetwork
+
+DEFAULT_REF_TEMPLATE = '#/definitions/{model}'
```

Comment:
I'm happy to change it, but do you have a reference changing it to `defs`? All I can seem to find in [the official docs](https://json-schema.org/draft/2020-12/json-schema-core.html#name-schema-re-use-with-defs) uses `$defs` rather than just `defs`. Either way is fine with me, just want to do it "right".

Reply:
I will probably keep this set as `definitions` for a little while to reduce the size of the diff of the tests, but I do agree we should change it to `$defs`